frontend/src/metabase/query_builder/components/filters/modals/BulkFilterItem/BulkFilterItem.tsx

```diff
@@ -94,10 +105,17 @@ export const BulkFilterItem = ({
 };
 
 const getNewFilter = (query: StructuredQuery, dimension: Dimension): Filter => {
-  const filter = new Filter([], null, dimension.query() ?? query);
-  const isBooleanField = isBoolean(dimension.field());
+  let filter = new Filter([], null, dimension.query() ?? query);
+  const field = dimension.field();
+  const isBooleanField = isBoolean(field);
+  const isTextField = isString(field) && field.has_field_values !== "list";
```

Comment:
Is it possible to move this closer to where it's gonna be used (line 117)? I'm not sure if that'll work as `filter` is reassigned at line 113. But my brief testing, it seems fine.